### ISSUE TITLE
Xamarin.Forms Previewer : NullReference Exception with Shell

### ISSUE THREAD

**Issue by HakanCelikMK1** [xamarin/Xamarin.Forms#5908]
### Description
I’ve created a project in Visual Studio 2019 from Mobile App (Xamarin.Forms) template with Shell and I built the project.
When I tried to open Xamarin.Forms Previewer by double clicking on AppShell.xaml, I got an exception message at the top of the previewer and the previewer itself was empty.

I've also reported the issue to Visual Studio Developer Community :
[Visual Studio 2019 16.0.1 - Xamarin.Forms Previewer : NullReference Exception with Shell](https://developercommunity.visualstudio.com/content/problem/531649/visual-studio-2019-1601-xamarinforms-previewer-nul.html)

### Steps to Reproduce
From Visual Studio 2019 16.0.1

1. New Project
2. Select Mobile App (Xamarin.Forms). I named it XFCTest01.
3. New Cross Platform App -> Shell template -> Platform : Android, iOS - Mobile Backend : No
4. Configuration Manager : iOS Project -> Build : No, Deploy : No
5. Build -> Rebuild Solution
6. Double click XFCTest01 -> AppShell.xaml. Xamarin.Forms Previewer : An exceptiom occurred while rendering the control. System.NullReferenceException is thrown.

### Expected Behavior
The page should have been rendered correctly.

### Actual Behavior
NullReference Exception is thrown with an empty previewer.

From AndroidDesigner*.log
```
The renderer encountered an error while rendering this file: System.NullReferenceException: Object reference not set to an instance of an object
Xamarin.Forms.Platform.Android.ShellFlyoutTemplatedContentRenderer.LoadView (Xamarin.Forms.Platform.Android.IShellContext shellContext) in D:\a\1\s\Xamarin.Forms.Platform.Android\Renderers\ShellFlyoutTemplatedContentRenderer.cs:47
Xamarin.Forms.Platform.Android.ShellFlyoutTemplatedContentRenderer..ctor (Xamarin.Forms.Platform.Android.IShellContext shellContext) in D:\a\1\s\Xamarin.Forms.Platform.Android\Renderers\ShellFlyoutTemplatedContentRenderer.cs:35
Xamarin.Forms.Platform.Android.ShellRenderer.CreateShellFlyoutContentRenderer () in D:\a\1\s\Xamarin.Forms.Platform.Android\Renderers\ShellRenderer.cs:163
Xamarin.Forms.Platform.Android.ShellRenderer.Xamarin.Forms.Platform.Android.IShellContext.CreateShellFlyoutContentRenderer () in D:\a\1\s\Xamarin.Forms.Platform.Android\Renderers\ShellRenderer.cs:89
Xamarin.Forms.Platform.Android.ShellFlyoutRenderer.AttachFlyout (Xamarin.Forms.Platform.Android.IShellContext context, Android.Views.View content) in D:\a\1\s\Xamarin.Forms.Platform.Android\Renderers\ShellFlyoutRenderer.cs:86
Xamarin.Forms.Platform.Android.ShellFlyoutRenderer.Xamarin.Forms.Platform.Android.IShellFlyoutRenderer.AttachFlyout (Xamarin.Forms.Platform.Android.IShellContext context, Android.Views.View content) in D:\a\1\s\Xamarin.Forms.Platform.Android\Renderers\ShellFlyoutRenderer.cs:20
Xamarin.Forms.Platform.Android.ShellRenderer.OnElementSet (Xamarin.Forms.Shell shell) in D:\a\1\s\Xamarin.Forms.Platform.Android\Renderers\ShellRenderer.cs:222
Xamarin.Forms.Platform.Android.ShellRenderer.Xamarin.Forms.Platform.Android.IVisualElementRenderer.SetElement (Xamarin.Forms.VisualElement element) in D:\a\1\s\Xamarin.Forms.Platform.Android\Renderers\ShellRenderer.cs:52
Xamarin.Forms.Platform.Android.Platform.CreateRenderer (Xamarin.Forms.VisualElement element, Android.Content.Context context) in D:\a\1\s\Xamarin.Forms.Platform.Android\Platform.cs:341
Xamarin.Forms.Platform.Android.Platform.CreateRenderer (Xamarin.Forms.VisualElement element) in D:\a\1\s\Xamarin.Forms.Platform.Android\Platform.cs:334
<unknown method>
System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture)
```
Also, there are three other exceptions before in the log which seem relevant :

```
[2019-04-13 09:20:47.4] Renderer >> INFO: Trying to load class xamarindesigner.AndroidFormsPreviewerRenderer
[2019-04-13 09:20:47.4] Renderer >> 
[2019-04-13 09:20:47.9] Renderer >> 4 [monodroid] SWITCHING CONTEXT
[2019-04-13 09:20:47.9] Renderer >> 
[2019-04-13 09:20:47.9] Renderer >> =================================================================
[2019-04-13 09:20:47.9] Renderer >> 	Native Crash Reporting
[2019-04-13 09:20:47.9] Renderer >> =================================================================
[2019-04-13 09:20:47.9] Renderer >> Got a SIGSEGV while executing native code. This usually indicates
[2019-04-13 09:20:47.9] Renderer >> a fatal error in the mono runtime or one of the native libraries 
[2019-04-13 09:20:47.9] Renderer >> used by your application.
[2019-04-13 09:20:47.9] Renderer >> =================================================================
```

```
[2019-04-13 09:20:49.2] Renderer >> INFO: Trying to load class android.support.design.widget.CoordinatorLayout
[2019-04-13 09:20:49.2] Renderer >> 
[2019-04-13 09:20:49.2] Renderer >> ERROR: An exception occured while loading control android.support.design.widget.CoordinatorLayout
[2019-04-13 09:20:49.2] Renderer >> ERROR: java.lang.reflect.InvocationTargetException
...
[2019-04-13 09:20:49.2] Renderer >> Caused by: java.lang.IllegalArgumentException: You need to use a Theme.AppCompat theme (or descendant) with the design library.
[2019-04-13 09:20:49.2] Renderer >> 	at android.support.design.widget.ThemeUtils.checkAppCompatTheme(ThemeUtils.java:33)
[2019-04-13 09:20:49.2] Renderer >> 	at android.support.design.widget.CoordinatorLayout.<init>(CoordinatorLayout.java:206)
[2019-04-13 09:20:49.2] Renderer >> 	at android.support.design.widget.CoordinatorLayout.<init>(CoordinatorLayout.java:200)
[2019-04-13 09:20:49.2] Renderer >> 	... 36 more
```

```
[2019-04-13 09:20:49.3] Renderer >> INFO: Trying to load class android.support.design.widget.AppBarLayout
[2019-04-13 09:20:49.3] Renderer >> 
[2019-04-13 09:20:49.3] Renderer >> ERROR: An exception occured while loading control android.support.design.widget.AppBarLayout
[2019-04-13 09:20:49.3] Renderer >> ERROR: java.lang.reflect.InvocationTargetException
...
[2019-04-13 09:20:49.3] Renderer >> Caused by: java.lang.IllegalArgumentException: You need to use a Theme.AppCompat theme (or descendant) with the design library.
[2019-04-13 09:20:49.3] Renderer >> 	at android.support.design.widget.ThemeUtils.checkAppCompatTheme(ThemeUtils.java:33)
[2019-04-13 09:20:49.3] Renderer >> 	at android.support.design.widget.AppBarLayout.<init>(AppBarLayout.java:158)
[2019-04-13 09:20:49.3] Renderer >> 	... 40 more
```

### Basic Information

- Version with issue: 
  Xamarin Designer 4.17.4.418 (remotes/origin/d16-0@3d086e814)
  Xamarin.Forms 3.6.0.293080
- Last known good version: Unknown
- IDE: Microsoft Visual Studio Community 2019 Version 16.0.1 (16.0.1+28803.156)
- Platform Target Frameworks: 
  - Android: 8.1
- Android Support Library Version: 28.0.0.1

### Software Environment
Microsoft Windows 10 Enterprise - Version 10.0.17134 Build 17134
Microsoft .NET Framework Version 4.7.03056

Microsoft Visual Studio Community 2019 Version 16.0.1
VisualStudio.16.Release/16.0.1+28803.156

Xamarin   16.0.0.515 (d16-0@a3502d929)
Xamarin Designer   4.17.4.418 (remotes/origin/d16-0@3d086e814)
Xamarin Templates   16.1.47 (31f664d)
Xamarin.Android SDK   9.2.0.5 (HEAD/292d27fa)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: mono/mono/2018-08-rc@5ad371dab1b
    Java.Interop: xamarin/java.interop/d16-0@c987483
    LibZipSharp: grendello/LibZipSharp/master@44de300
    LibZip: nih-at/libzip/rel-1-5-1@b95cf3f
    MXE: xamarin/mxe/xamarin@b9cbb535
    ProGuard: xamarin/proguard/master@905836d
    SQLite: xamarin/sqlite/3.26.0@325e91a
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-0@0a7edd6
Xamarin.iOS and Xamarin.Mac SDK   12.8.0.0 (0aa8452)

Android SDK Tools 26.1.1
Android SDK Platform Tools 28.0.0
Android SDK Build-Tools 27.0.3
Android 8.1 Oreo - Android SDK Platform 27 v3
Microsoft OpenJDK 1.8.0-25

I also have VS 2017 installed on my machine if it makes any difference
Microsoft Visual Studio Community 2017 Version 15.9.11
VisualStudio.15.Release/15.9.11+28307.586

### Reproduction Project and Logs

I'm attaching the project and Xamarin logs.

[XFCTest01.zip](https://github.com/xamarin/Xamarin.Forms/files/3076104/XFCTest01.zip)

[VS2019_XamarinLogs.zip](https://github.com/xamarin/Xamarin.Forms/files/3076103/VS2019_XamarinLogs.zip)

> VS bug [#847193](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/847193)

**Comment by a-jiang** [comment]
I encountered the same exception. Didn't they test it?

**Comment by HakanCelikMK1** [comment]
@a-jiang Do you also have VS 2017 on your machine ? I was thinking, perhaps it was an unintended consequence of installing both VS 2019 and VS 2017 on the same machine.

**Comment by a-jiang** [comment]
Yes,both VS 2017 and VS 2019 are installed in my machine.But I'm not sure if that's the reason.Should I uninstall VS 2017? It's really very strange. I hope they can find and fix the bug as soon as possible.

**Comment by a-jiang** [comment]
@HakanCelikMK1  Yes,both VS 2017 and VS 2019 are installed in my machine.But I'm not sure if that's the reason.Should I uninstall VS 2017? It's really very strange. I hope they can find and fix the bug as soon as possible.

**Comment by samhouts** [comment]
closed by #5955
